source-braintree-native/source_braintree_native/api.py

```diff
@@ -108,36 +132,134 @@ async def snapshot_resources(
         yield FullRefreshResource.model_validate(_braintree_object_to_dict(object))
 
 
+async def _fetch_transaction_ids_for_single_field(
+    field: str,
+    braintree_gateway: BraintreeGateway,
+    start: datetime,
+    end: datetime,
+    log: Logger,
+) -> list[str]:
+
+    search_criteria: Search.RangeNodeBuilder = getattr(TransactionSearch, field)
+
+    collection: ResourceCollection = await asyncio.to_thread(
+        braintree_gateway.transaction.search,
+        search_criteria.between(start, end),
+    )
+
+    if collection.maximum_size >= TRANSACTION_SEARCH_LIMIT:
+        raise RuntimeError(_search_limit_error_message(collection.maximum_size, "transactions"))
+
+    return collection.ids
+
+
+async def _fetch_unique_updated_transaction_ids(
+    braintree_gateway: BraintreeGateway,
+    start: datetime,
+    end: datetime,
+    log: Logger,
+) -> list[str]:
+    async with asyncio.TaskGroup() as tg:
```

Comment:
I think this could be simplified using `asyncio.gather`

Reply:
So something more like this?
```python
tasks: list[Task[list[str]]] = []
for field in TRANSACTION_SEARCH_FIELDS:
    task = asyncio.create_task(
        _fetch_transaction_ids_for_single_field(
            field,
            braintree_gateway,
            start,
            end,
            log,
        )
    )

    tasks.append(task)

await asyncio.gather(*tasks)
```

Is `asyncio.gather` preferred here because we don't need all the extra bells & whistles around task cancellation that come with `asyncio.TaskGroup`?

Reply:
`asyncio.gather` is just a more succinct way to express what is happening. It's basically a helper for this common pattern. Can you do something like [this](https://github.com/estuary/connectors/blob/027f511b7deedfdbb19fd641244f6827f8a985d2/source-hubspot-native/source_hubspot_native/api.py#L220-L225)?

Reply:
Got it, I see now. Yep, I can do that.